core/lib/spree/app_configuration.rb

```diff
@@ -202,6 +202,10 @@ class AppConfiguration < Preferences::Configuration
     #   @return [Integer] Promotions to show per-page in the admin (default: +15+)
     preference :promotions_per_page, :integer, default: 15
 
+    # @!attribute [rw] disable_actionless_promotion_validation
+    #   @return [Boolean] Promotions should have actions associated before activation (default: +true+)
+    preference :actionless_promotion_inactive, :boolean, default: true
```

Comment:
I didn't really review the PR yet but in the meantime, I think this should be changed. The default value should be the one that represents the old behavior (`false` in this case). This way, users that upgrade Solidus will keep having the same behavior and will see a deprecation warning. They will have the chance to take a look at what happens before switching the value.

New Solidus users will have the new version of `spree.rb` initializer instead and they will start their apps with the new default. 

How does it sound?

Reply:
No worries and sounds fine. This will probably take me a little while as the change of function appears to break a lot of tests, so I need to reconcile the change across functions.